client/app/src/main/java/com/microsoft/research/karya/data/service/MicroTaskAPI.kt

```diff
@@ -1,32 +1,25 @@
 package com.microsoft.research.karya.data.service
 
+import com.google.gson.JsonArray
 import com.microsoft.research.karya.data.model.karya.KaryaFileRecord
 import okhttp3.MultipartBody
 import okhttp3.ResponseBody
+import org.json.JSONArray
 import retrofit2.Response
-import retrofit2.http.GET
-import retrofit2.http.Header
-import retrofit2.http.Multipart
-import retrofit2.http.POST
-import retrofit2.http.Part
-import retrofit2.http.Path
+import retrofit2.http.*
 
 interface MicroTaskAPI {
 
-    @Multipart
-    @POST("/microtask_assignment/{id}/output_file")
-    suspend fun postUploads(
-        @Header("auth-provider") authProvider: String,
+    @PUT("/assignments")
+    suspend fun submitAssignments(
         @Header("id-token") idTokenHeader: String,
-        @Path("id") microtaskAssignmentID: String,
-        @Part json: MultipartBody.Part,
-        @Part file: MultipartBody.Part
-    ): Response<KaryaFileRecord>
+        @Body updates: JsonArray
+    ): Response<JsonArray>
 
-    @GET("/microtask_assignment/{id}/input_file")
-    suspend fun getInputFileForAssignment(
-        @Header("auth-provider") authProvider: String,
-        @Header("id-token") idToken: String,
-        @Path("id") microtaskAssignmentID: String
-    ): Response<ResponseBody>
+    @GET("/assignments")
+    suspend fun getAssignments(
+        @Header("id-token") idTokenHeader: String,
+        @Query("type") type: String, //TODO: Make this an enum class
+        @Query("from") from: String,
+    ): Response<JSONArray> // TODO: Modify this output type according to response type
```

Comment:
If we know the server response already why is this still a `TODO`?

Reply:
Yeah, It was made when the API development was in progress. I will update this